modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ExtensionMethods.cs

```diff
@@ -149,39 +149,52 @@ public static string GetDeclarationKeyword(this INamedTypeSymbol namedTypeSymbol
             };
         }
 
-        private static SymbolDisplayFormat FullyQualifiedFormatOmitGlobal { get; } =
-            SymbolDisplayFormat.FullyQualifiedFormat
-                .WithGlobalNamespaceStyle(SymbolDisplayGlobalNamespaceStyle.Omitted);
-
-        public static string FullQualifiedName(this ITypeSymbol symbol)
-            => symbol.ToDisplayString(NullableFlowState.NotNull, FullyQualifiedFormatOmitGlobal);
-
         public static string NameWithTypeParameters(this INamedTypeSymbol symbol)
         {
             return symbol.IsGenericType ?
                 string.Concat(symbol.Name, "<", string.Join(", ", symbol.TypeParameters), ">") :
                 symbol.Name;
         }
 
-        public static string FullQualifiedName(this INamespaceSymbol namespaceSymbol)
+        private static SymbolDisplayFormat FullyQualifiedFormatOmitGlobal { get; } =
+            SymbolDisplayFormat.FullyQualifiedFormat
+                .WithGlobalNamespaceStyle(SymbolDisplayGlobalNamespaceStyle.Omitted);
+
+        private static SymbolDisplayFormat FullyQualifiedFormatIncludeGlobal { get; } =
+            SymbolDisplayFormat.FullyQualifiedFormat
+                .WithGlobalNamespaceStyle(SymbolDisplayGlobalNamespaceStyle.Included);
+
+        public static string FullQualifiedNameOmitGlobal(this ITypeSymbol symbol)
+            => symbol.ToDisplayString(NullableFlowState.NotNull, FullyQualifiedFormatOmitGlobal);
+
+        public static string FullQualifiedNameOmitGlobal(this INamespaceSymbol namespaceSymbol)
             => namespaceSymbol.ToDisplayString(FullyQualifiedFormatOmitGlobal);
 
-        public static string FullQualifiedName(this ISymbol symbol)
+        public static string FullQualifiedNameOmitGlobal(this ISymbol symbol)
             => symbol.ToDisplayString(FullyQualifiedFormatOmitGlobal);
 
+        public static string FullQualifiedNameIncludeGlobal(this ITypeSymbol symbol)
+            => symbol.ToDisplayString(NullableFlowState.NotNull, FullyQualifiedFormatIncludeGlobal);
+
+        public static string FullQualifiedNameIncludeGlobal(this INamespaceSymbol namespaceSymbol)
+            => namespaceSymbol.ToDisplayString(FullyQualifiedFormatIncludeGlobal);
+
+        public static string FullQualifiedNameIncludeGlobal(this ISymbol symbol)
+            => symbol.ToDisplayString(FullyQualifiedFormatIncludeGlobal);
```

Comment:
I'm not a fan of this. When calling this method, an `ITypeSymbol` would result in a different result if upcast to `ISymbol`.

Reply:
Me neither. These overloads already existed, the one that takes `ISymbol` was added recently but maybe it can be removed, would that be enough? I think the other overloads are less likely to be an issue.

Reply:
It seems to be used only by `FullQualifiedSyntax`, so we can move it there as a local static method, and maybe change the name.

Reply:
I removed it, now `FullQualifiedSyntax` explicitly calls `ToDisplayString` with the appropiate `SymbolDisplayFormat`.